cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h

```diff
@@ -291,6 +291,40 @@ class SPCORE_API Unreal
         return createComponentOutsideOwnerConstructor<TSceneComponent, TReturnAsSceneComponent>(owner, owner, scene_component_name);
     }
 
+    static UActorComponent* createComponentOutsideOwnerConstructorByClass(UClass* component_class, AActor* owner, const std::string& component_name)
```

Comment:
- Add `createComponentInsideOwnerConstructorByClass` variants for consistency.
- Make the templated variants call the non-templated variants to reduce code duplication.